src/trajectory.py

```diff
@@ -0,0 +1,110 @@
+import pickle
+from pathlib import Path
+from typing import Optional
+
+from pymatgen.core import Lattice
+from pymatgen.core.trajectory import Trajectory as PymatgenTrajectory
+from pymatgen.io import vasp
+
+
+class Trajectory(PymatgenTrajectory):
```

Comment:
I noticed that PymatgenTrajectory has no object variables defined inside the class (like a dataclass or a pydantic object), when we merge this back we might introduce that for them as well. Lets create an issue for this.